qaequilibrae/modules/processing_provider/MatrixFromLayer.py

```diff
@@ -0,0 +1,135 @@
+                    QgsProcessingParameterString('CoreName', self.tr('Name for the core of your matrix'), multiLine=False, defaultValue='Value')
+                    ]
+        for param in advparams:
+            param.setFlags(param.flags() | QgsProcessingParameterDefinition.FlagAdvanced)
+            self.addParameter(param)
+
+    def processAlgorithm(self, parameters, context, model_feedback):
+        feedback = QgsProcessingMultiStepFeedback(3, model_feedback)
+        results = {}
+        outputs = {}
+        
+        # Checks if we have access to aequilibrae library
+        has_aeq = iutil.find_spec("aequilibrae") is not None
+        
+        if not has_aeq:
+            sys.exit(tr('AequilibraE library not found'))
+        from aequilibrae.matrix import AequilibraeMatrix
+        
+        origin=parameters['origin']
+        destination=parameters['destination']
+        value=parameters['value']
+        
+        matrix_name=parameters['MatrixName']
+        matrix_description=parameters['MatrixDescription']
+        core_name=[parameters['CoreName']]
+        
+        output_folder=parameters['OutputFold']
+        output_name=parameters['FileName']
+        filename=os.path.join(output_folder,output_name+'.aem')
+        
+        # Import layer as a pandas df
+        feedback.pushInfo(tr('Importing the layer from QGIS :'))
+        layer = self.parameterAsVectorLayer(parameters, 'matrix_layer', context)
+        cols=[origin,destination,value]
+        datagen = ([f[col] for col in cols] for f in layer.getFeatures())
+        matrix = pd.DataFrame.from_records(data=datagen, columns=cols)
+        feedback.pushInfo(str(matrix.head(5)))
+        feedback.pushInfo('...')
+        feedback.pushInfo('')
+        feedback.setCurrentStep(1)
+        
+        # Getting all zones
+        all_zones = np.array(sorted(list(set( list(matrix[origin].unique()) + list(matrix[destination].unique())))))
+        num_zones = all_zones.shape[0]
+        idx = np.arange(num_zones)
+
+        # Creates the indexing dataframes
+        origs = pd.DataFrame({"from_index": all_zones, "from":idx})
+        dests = pd.DataFrame({"to_index": all_zones, "to":idx})
+
+        # adds the new index columns to the pandas dataframe
+        matrix = matrix.merge(origs, left_on=origin, right_on='from_index', how='left')
+        matrix = matrix.merge(dests, left_on=destination, right_on='to_index', how='left')
+
+        agg_matrix = matrix.groupby(['from', 'to']).sum()
+
+        # returns the indices
+        agg_matrix.reset_index(inplace=True)
+
+        # Creating the aequilibrae matrix file
+        mat = AequilibraeMatrix()
+        mat.name=matrix_name
+        mat.description=matrix_description
+
+        kwargs = {'file_name': filename,
+                  'zones': num_zones,
+                  'matrix_names': core_name,
+                  'memory_only': False}
+
+        mat.create_empty(**kwargs)
```

Comment:
Was there a reason to not just use keywords directly?

```suggestion
        mat.create_empty(file_name = filename,
                         zones = num_zones,
                         matrix_names = core_name,
                         memory_only = False)
```